crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -1014,136 +820,30 @@ mod tests {
-                .build(sender.clone());
-
-            assert_eq!(*list.inner.ranges.read().unwrap(), &[0..=1, 2..=3]);
-
-            list.set_range(4..=5).unwrap();
-            assert_eq!(*list.inner.ranges.read().unwrap(), &[4..=5]);
-        }
-
-        // Set range on `Growing`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_growing(20, None))
-                .build(sender.clone());
-
-            assert!(list.set_range(4..=5).is_err());
-        }
-
-        // Set range on `Paging`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_paging(20, None))
-                .build(sender);
-
-            assert!(list.set_range(4..=5).is_err());
-        }
-    }
-
-    #[test]
-    fn test_sliding_sync_list_add_range() {
-        let (sender, _receiver) = channel(1);
-
-        // Add range on `Selective`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_selective())
-                .ranges(vec![0..=1])
-                .build(sender.clone());
-
-            assert_eq!(*list.inner.ranges.read().unwrap(), &[0..=1]);
-
-            list.add_range(2..=3).unwrap();
-            assert_eq!(*list.inner.ranges.read().unwrap(), &[0..=1, 2..=3]);
-        }
-
-        // Add range on `Growing`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_growing(20, None))
-                .build(sender.clone());
-
-            assert!(list.add_range(2..=3).is_err());
-        }
-
-        // Add range on `Paging`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_paging(20, None))
-                .build(sender);
-
-            assert!(list.add_range(2..=3).is_err());
-        }
-    }
-
-    #[test]
-    fn test_sliding_sync_list_reset_ranges() {
-        let (sender, _receiver) = channel(1);
-
-        // Reset ranges on `Selective`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_selective())
-                .ranges(vec![0..=1])
-                .build(sender.clone());
-
-            assert_eq!(*list.inner.ranges.read().unwrap(), &[0..=1]);
-
-            list.reset_ranges().unwrap();
-            assert!(list.inner.ranges.read().unwrap().is_empty());
-        }
-
-        // Reset range on `Growing`.
-        {
-            let list = SlidingSyncList::builder("foo")
-                .sync_mode(SlidingSyncMode::new_growing(20, None))
-                .build(sender.clone());
-
-            assert!(list.reset_ranges().is_err());
-        }
+        let list = SlidingSyncList::builder("foo")
+            .sync_mode(
+                SlidingSyncSelectiveModeBuilder::new().add_range(0..=1).add_range(2..=3).build(),
+            )
+            .build(sender);
```

Comment:
We should add a test to ensure that `sync_mode` emits a message in the internal channel. I can do that if you want. (Here, it's just a matter of testing `_receiver`).